src/hostcalls.rs

```diff
@@ -1030,6 +1030,9 @@ pub fn call_foreign_function(
                 }
             }
             Status::NotFound => Err(Status::NotFound),
```

Comment:
optional: could using a multi-pattern match avoid some repetition, i.e.

```
status @ (Status::NotFound | Status::BadArgument | Status::SerializationFailure | Status::InternalFailure) =>
  Err(status)
```

Reply:
I find the multiple cases much more readable, especially compared to very long lines.